test/common.js

```diff
@@ -2,31 +2,53 @@ const Promise = require('bluebird');
 const { Readable } = require('stream');
 const _ = require('lodash');
 const superagent = require('superagent');
+const Broker = require('..');
+const { AssertionError } = require('chai');
 
 const API_URL = 'http://localhost:15672/api';
 const API_AUTH_ARGS = ['guest', 'guest'];
 
 module.exports = {
-    async cleanup(broker, exchanges, ...queues) {
+    /**
+     * @param [broker]
+     */
+    async cleanup(broker) {
         if (process.env.NO_TEST_CLEANUP)
             return;
 
-        exchanges = [].concat(exchanges);
-        let channel = await(await broker.getConnection()).createChannel();
+        if (broker)
+            broker.disconnect();
 
-        console.log(`=============================== CLEANUP: deleting queues ===============================`);
-        await Promise.each(queues, async q => {
-            if (q)
-                await channel.deleteQueue(q);
-        });
+        await this.deleteAllViaApi('queues', { ignoreAutoGenerated: false });
+        await this.deleteAllViaApi('exchanges', { ignoreAutoGenerated: true });
+        console.log(`=============================== CLEANUP: finished ===============================`);
+    },
+
+    async getFromApi(...pathParts) {
+        let response = await superagent.get(`${API_URL}/${_.join(pathParts, '/')}`).auth(...API_AUTH_ARGS);
+        return response.body;
+    },
+
+    async deleteAllViaApi(entityType, { addDefaultVhost = true, ignoreAutoGenerated = true } = {}) {
+        let entities = await this.getFromApi(entityType);
 
-        console.log(`=============================== CLEANUP: deleting exchanges ===============================`);
-        await Promise.each(exchanges, async x => {
-            if (x)
-                await channel.deleteExchange(x);
+        if (ignoreAutoGenerated)
+            entities = entities.filter(x => x && x.name && x.name.indexOf('amq.') === -1);
+
+        console.log(`=============================== CLEANUP: deleting ${entities.length} ${entityType} ===============================`);
+
+        await Promise.each(entities, async entity => {
+            await this.deleteViaApi(entityType, entity.name, { addDefaultVhost });
         });
+    },
 
-        console.log(`=============================== CLEANUP: finished ===============================`);
+    async deleteViaApi(entityType, name, { addDefaultVhost = true } = {}) {
+        let vhost = addDefaultVhost ? '%2F/' : '';
```

Comment:
I still maintain that `%2F/` is unreadable to the average developer :)
```suggestion
        let vhost = addDefaultVhost ? encodeURIComponent('/')+'/' : '';
```